stdlib/LibGit2/src/repository.jl

```diff
@@ -142,7 +142,7 @@ function (::Type{T})(repo::GitRepo, spec::AbstractString) where T<:GitObject
     @check ccall((:git_revparse_single, :libgit2), Cint,
                  (Ptr{Ptr{Cvoid}}, Ptr{Cvoid}, Cstring), obj_ptr_ptr, repo.ptr, spec)
     # check object is of correct type
-    if T != GitObject && T != GitUnknownObject
+    if T !== GitObject && T !== GitUnknownObject
```

Comment:
```suggestion
    if !(T <: GitObject || T <: GitUnknownObject)
```
might convey the intention better?